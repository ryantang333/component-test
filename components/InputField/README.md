# @test/input-field

Welcome to my `InputField` component.

## Installation

```sh
npm i @test/input-field
# or with yarn
yarn add @test/input-field
```

Then to use the component in your code just import it!

```js
import InputField from '@test/input-field';
// and with css-modules
import '@test/input-field/dist/main.css';
```
