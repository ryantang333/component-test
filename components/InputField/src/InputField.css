.input-field {
  height: 1.4375rem;
  padding: 1.25rem 0.875rem;
  width: 100%;
  box-sizing: border-box;
  border: none;
  border-radius: inherit;
}

.input-field:focus {
  outline: none;
}
