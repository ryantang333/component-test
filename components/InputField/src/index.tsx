import React from 'react';
import styles from './InputField.css';

interface InputFieldProps {
  /** Default value */
  defaultValue?: string;
  /** Disabled */
  disabled?: boolean;
  /** ID for input and corresponding label */
  id?: string;
  /** The onChange function */
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  /** Read only */
  readOnly?: boolean;
  /** Marks input as required (adds an asterisk) */
  required?: boolean;
  /** User-provided styles */
  style?: React.CSSProperties;
  /** Input type */
  type?: string;
}

/** A simple component. */
export const InputField = (props: InputFieldProps) => {
  const {
    defaultValue,
    disabled,
    id,
    onChange,
    readOnly,
    required,
    style,
    type = 'text'
  } = props;

  return (
    <input
      className={styles.inputField}
      defaultValue={defaultValue}
      disabled={disabled}
      id={id}
      readOnly={readOnly}
      required={required}
      style={style}
      type={type}
      onChange={onChange}
    />
  );
};

export default InputField;
