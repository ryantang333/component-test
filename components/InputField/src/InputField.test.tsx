import React from 'react';
import { render } from '@testing-library/react';
import InputField from '.';

describe('InputField', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<InputField />);
    expect(container).toMatchSnapshot();
  });
});
