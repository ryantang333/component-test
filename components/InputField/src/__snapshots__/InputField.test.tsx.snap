// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField It matches the snapshot 1`] = `
<div>
  <input
    class="inputField"
    type="text"
    value=""
  />
</div>
`;
